spec/unit/pdk/module/template_dir_spec.rb

```diff
@@ -174,4 +174,47 @@
       end
     end
   end
+
+  describe '.config_for(dest_path)' do
+    before(:each) do
+      allow(File).to receive(:directory?).with(anything).and_return(true)
+      allow(PDK::Util).to receive(:make_tmpdir_name).with('pdk-module-template').and_return('/tmp/path')
+      allow(PDK::CLI::Exec).to receive(:git).with('clone', path_or_url, '/tmp/path').and_return(exit_code: 0)
+      allow(File).to receive(:file?).with(anything).and_return(File.join(path_or_url, 'config_defaults.yml')).and_return(true)
+      allow(File).to receive(:read).with(File.join(path_or_url, 'config_defaults.yml')).and_return(config_defaults)
+    end
+    context 'when the module has a .sync.yml file' do
+      let(:yaml_text) do
+        <<-EOF
+       appveyor.yml:
+         delete: true
+       .travis.yml:
+         extras:
+         - rvm: 2.1.9
+       foo:
+         attr:
+         - val: 3
+       EOF
+      end
+      let(:yaml_hash) do
+        YAML.load(yaml_text) # rubocop:disable Security/YAMLLoad
+      end
+      let(:config_hash) do
+        YAML.load(config_defaults) # rubocop:disable Security/YAMLLoad
+      end
+
+      before(:each) do
```

Comment:
The `config_defaults` related allows should be moved up to the top level `before` block for maintenance purposes since they are required to be mocked with or without a `sync.yml` file.